### PR TITLE
Build system

### PR DESCRIPTION
**Pull request by SDAChess**
This pull request introduces a lot of changes in the build system, division into git submodules a better install rule.

This is a starting point and I think some things could be improved but the overall structure is better.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -90,109 +155,49 @@ if (ARK_ENABLE_SYSTEM)
...
 # linking libraries
-
 find_package(Threads)
 
 if (UNIX OR LINUX)
-    if (CMAKE_COMPILER_IS_GNUCXX)
-        target_link_libraries(ArkReactor PUBLIC stdc++fs)
-    elseif (CMAKE_COMPILER_IS_CLANG)
-        target_link_libraries(ArkReactor PUBLIC stdc++fs)
-    endif()
```
Comment (SuperFola) [inline CMakeLists.txt:142]: this is actually needed, at least on Ubuntu, otherwise it can't compile

**Review by SuperFola — CHANGES_REQUESTED** [review]

**Comment by SuperFola** [comment]
Also when reading the CI logs, seems like we need a -fPIC in the ArkString: https://github.com/ArkScript-lang/Ark/pull/306/checks?check_run_id=3660188801#step:6:62

**Comment by SDAChess** [comment]
Ok this is worse than I expected, I don't know why there is so many errors. I'll try to fix this one today.

File: CMakeLists.txt [hunk 2]
```diff
@@ -43,21 +120,7 @@ if (LinkerFlags)
     set(CMAKE_EXE_LINKER_FLAGS "${LinkerFlags}" CACHE STRING "Linker options" FORCE)
 endif()
 
-set(CMAKE_POSITION_INDEPENDENT_CODE ON)
```
Comment (SuperFola) [inline CMakeLists.txt:46]: I would suggest adding this back

**Review by SuperFola — APPROVED** [review]

**Pull request merged**